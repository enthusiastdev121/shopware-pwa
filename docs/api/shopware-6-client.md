<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [@shopware-pwa/shopware-6-client](./shopware-6-client.md)

## shopware-6-client package

## Functions

|  Function | Description |
|  --- | --- |
|  [getCustomer()](./shopware-6-client.getcustomer.md) | <b><i>(BETA)</i></b> Get customer's object |
|  [getCustomerAddresses()](./shopware-6-client.getcustomeraddresses.md) | <b><i>(BETA)</i></b> Get all customer's addresses |
|  [getCustomerOrders()](./shopware-6-client.getcustomerorders.md) | <b><i>(BETA)</i></b> Get all customer's orders |
|  [invokeGet({ address })](./shopware-6-client.invokeget.md) | <b><i>(BETA)</i></b> Invoke custom GET request to shopware API. Mostly for plugins usage. You can skip domain and pass only endpoint ex. <code>/api/my/endpoint</code> |
|  [invokePost({ address, payload, })](./shopware-6-client.invokepost.md) | <b><i>(BETA)</i></b> Invoke custom POST request to shopware API. Mostly for plugins usage. You can skip domain and pass only endpoint ex. <code>/api/my/endpoint</code> |
|  [login({ username, password, })](./shopware-6-client.login.md) | <b><i>(BETA)</i></b> Login user to shopware instance. |
|  [logout()](./shopware-6-client.logout.md) | <b><i>(BETA)</i></b> End up the user session. |
|  [onConfigChange(fn)](./shopware-6-client.onconfigchange.md) | <b><i>(BETA)</i></b> |
|  [setup(config)](./shopware-6-client.setup.md) | <b><i>(BETA)</i></b> Setup configuration. Merge default values with provided in param. This method will override existing config. For config update invoke \*\*update\*\* method. |
|  [update(config)](./shopware-6-client.update.md) | <b><i>(BETA)</i></b> Update current configuration. This will change only provided values. |

## Interfaces

|  Interface | Description |
|  --- | --- |
|  [ClientSettings](./shopware-6-client.clientsettings.md) | <b><i>(BETA)</i></b> |
|  [ConfigChangedArgs](./shopware-6-client.configchangedargs.md) | <b><i>(BETA)</i></b> |

## Variables

|  Variable | Description |
|  --- | --- |
|  [config](./shopware-6-client.config.md) | <b><i>(BETA)</i></b> |

