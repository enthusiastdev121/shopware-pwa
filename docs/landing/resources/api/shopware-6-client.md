<!-- Do not edit this file. It is automatically generated by API Documenter. -->

[Home](./index.md) &gt; [@shopware-pwa/shopware-6-client](./shopware-6-client.md)

## shopware-6-client package

## Functions

|  Function | Description |
|  --- | --- |
|  [addCartItemQuantity(itemId, quantity, contextInstance)](./shopware-6-client.addcartitemquantity.md) | <b><i>(BETA)</i></b> Increases the current quantity in specific cart line item by given quantity.<!-- -->Example: If current quantity is 3 and you pass 2 as quantity parameter, you will get a new cart's state with quantity 5. |
|  [addProductToCart(productId, quantity, contextInstance)](./shopware-6-client.addproducttocart.md) | <b><i>(BETA)</i></b> Adds specific quantity of the product to the cart by productId. It creates a new cart line item.<!-- -->Warning: This method does not change the state of the cart in any way if productId already exists in a cart. For changing the quantity use addQuantityToCartLineItem() or changeCartLineItemQuantity() methods. |
|  [changeCartItemQuantity(itemId, newQuantity, contextInstance)](./shopware-6-client.changecartitemquantity.md) | <b><i>(BETA)</i></b> Changes the current quantity in specific cart line item to given quantity.<!-- -->Example: If current quantity is 3 and you pass 2 as quantity parameter, you will get a new cart's state with quantity 2. |
|  [createInstance(initialConfig)](./shopware-6-client.createinstance.md) | <b><i>(BETA)</i></b> |
|  [getCart(contextInstance)](./shopware-6-client.getcart.md) | <b><i>(BETA)</i></b> Gets the current cart for the sw-context-token. |
|  [getCustomer(contextInstance)](./shopware-6-client.getcustomer.md) | <b><i>(BETA)</i></b> Get customer's object |
|  [getCustomerAddresses(contextInstance)](./shopware-6-client.getcustomeraddresses.md) | <b><i>(BETA)</i></b> Get all customer's addresses |
|  [getCustomerOrders(contextInstance)](./shopware-6-client.getcustomerorders.md) | <b><i>(BETA)</i></b> Get all customer's orders |
|  [getOrderPaymentUrl({ orderId, finishUrl, }, contextInstance)](./shopware-6-client.getorderpaymenturl.md) | <b><i>(BETA)</i></b> Get payment address to redirect user after placing order. |
|  [getProductPage(path, searchCriteria, contextInstance)](./shopware-6-client.getproductpage.md) | <b><i>(BETA)</i></b> |
|  [getResults(term, searchCriteria, contextInstance)](./shopware-6-client.getresults.md) | <b><i>(BETA)</i></b> |
|  [getSearchResults(term, searchCriteria, contextInstance)](./shopware-6-client.getsearchresults.md) | <b><i>(BETA)</i></b> |
|  [getStoreNavigation({ requestActiveId, requestRootId, depth, buildTree, searchCriteria, }, contextInstance)](./shopware-6-client.getstorenavigation.md) | <b><i>(BETA)</i></b> |
|  [getSuggestedResults(term, searchCriteria, contextInstance)](./shopware-6-client.getsuggestedresults.md) | <b><i>(BETA)</i></b> |
|  [invokeGet({ address }, contextInstance)](./shopware-6-client.invokeget.md) | <b><i>(BETA)</i></b> Invoke custom GET request to shopware API. Mostly for plugins usage. You can skip domain and pass only endpoint ex. <code>/api/my/endpoint</code> |
|  [invokePost({ address, payload, }, contextInstance)](./shopware-6-client.invokepost.md) | <b><i>(BETA)</i></b> Invoke custom POST request to shopware API. Mostly for plugins usage. You can skip domain and pass only endpoint ex. <code>/api/my/endpoint</code> |
|  [login({ username, password }, contextInstance)](./shopware-6-client.login.md) | <b><i>(BETA)</i></b> Login user to shopware instance. |
|  [logout(contextInstance)](./shopware-6-client.logout.md) | <b><i>(BETA)</i></b> End up the user session. |
|  [newsletterSubscribe(params, contextInstance)](./shopware-6-client.newslettersubscribe.md) | <b><i>(BETA)</i></b> |
|  [newsletterUnsubscribe({ email, }, contextInstance)](./shopware-6-client.newsletterunsubscribe.md) | <b><i>(BETA)</i></b> |
|  [removeCartItem(itemId, contextInstance)](./shopware-6-client.removecartitem.md) | <b><i>(BETA)</i></b> Deletes the cart line item by id.<!-- -->This method may be used for deleting "product" type item lines as well as "promotion" type item lines. |
|  [sendContactForm(params, contextInstance)](./shopware-6-client.sendcontactform.md) | <b><i>(BETA)</i></b> |

## Interfaces

|  Interface | Description |
|  --- | --- |
|  [ClientSettings](./shopware-6-client.clientsettings.md) | <b><i>(BETA)</i></b> |
|  [ConfigChangedArgs](./shopware-6-client.configchangedargs.md) | <b><i>(BETA)</i></b> |
|  [ContactFormData](./shopware-6-client.contactformdata.md) | <b><i>(BETA)</i></b> |
|  [GetStoreNavigationParams](./shopware-6-client.getstorenavigationparams.md) | <b><i>(BETA)</i></b> More about the navigation parameters: https://docs.shopware.com/en/shopware-platform-dev-en/store-api-guide/navigation?category=shopware-platform-dev-en/store-api-guide |
|  [NewsletterSubscribeData](./shopware-6-client.newslettersubscribedata.md) | <b><i>(BETA)</i></b> |
|  [PageResolverProductResult](./shopware-6-client.pageresolverproductresult.md) | <b><i>(BETA)</i></b> |
|  [PageResolverResult](./shopware-6-client.pageresolverresult.md) | <b><i>(BETA)</i></b> |
|  [ShopwareApiInstance](./shopware-6-client.shopwareapiinstance.md) | <b><i>(BETA)</i></b> |

## Variables

|  Variable | Description |
|  --- | --- |
|  [config](./shopware-6-client.config.md) | <b><i>(BETA)</i></b> |
|  [onConfigChange](./shopware-6-client.onconfigchange.md) | <b><i>(BETA)</i></b> |
|  [setup](./shopware-6-client.setup.md) | <b><i>(BETA)</i></b> Setup configuration. Merge default values with provided in param. This method will override existing config. For config update invoke \*\*update\*\* method. |
|  [update](./shopware-6-client.update.md) | <b><i>(BETA)</i></b> Update current configuration. This will change only provided values. |

