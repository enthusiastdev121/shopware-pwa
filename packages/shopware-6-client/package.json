{
  "name": "@shopware-pwa/shopware-6-client",
  "version": "0.1.0-alpha.7",
  "description": "Rest API client for Shopware 6.",
  "main": "dist/shopware-6-client.cjs.js",
  "module": "dist/shopware-6-client.esm-bundler.js",
  "files": [
    "dist"
  ],
  "types": "dist/shopware-6-client.d.ts",
  "unpkg": "dist/shopware-6-client.global.js",
  "sideEffects": false,
  "repository": {
    "type": "git",
    "url": "git+https://github.com/DivanteLtd/shopware-pwa"
  },
  "buildOptions": {
    "name": "Shopware6Client",
    "formats": [
      "esm-bundler",
      "cjs"
    ]
  },
  "peerDependencies": {
    "axios": "^0.19.2",
    "https": "^1.0.0"
  },
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  }
}
