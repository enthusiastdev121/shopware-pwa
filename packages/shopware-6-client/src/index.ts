import { config, ClientSettings, setupConfig, updateConfig } from "./settings";
import { reloadConfiguration } from "./apiService";

export { config } from "./settings";

export * from "./services/categoryService";
export * from "./services/productService";
export * from "./services/customerService";
export * from "./services/contextService";
export * from "./services/cartService";
export * from "./services/navigationService";
export * from "./services/pageService";
export * from "./services/checkoutService";
export { ShopwareParams } from "./helpers/searchConverter";

/**
 * Setup configuration. Merge default values with provided in param.
 * This method will override existing config. For config update invoke **update** method.
 * @alpha
 */
export function setup(config: ClientSettings = {}): void {
  setupConfig(config);
  reloadConfiguration();
}

/**
 * Update current configuration. This will change only provided values.
 * @alpha
 */
export function update(config: ClientSettings = {}): void {
  updateConfig(config);
  reloadConfiguration();
  configChanged();
}

/**
 * @alpha
 */
export interface ConfigChangedArgs {
  config: ClientSettings;
}
const callbackMethods: ((context: ConfigChangedArgs) => void)[] = [];

/**
 * @alpha
 */
export function onConfigChange(fn: (context: ConfigChangedArgs) => void) {
  callbackMethods.push(fn);
}
function configChanged(): void {
  callbackMethods.forEach((fn) => fn({ config }));
}
