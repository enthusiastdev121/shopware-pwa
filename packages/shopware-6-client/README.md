# API client for Shopware 6

Compatibility with Shopware 6:  
[![shopware-ver](https://img.shields.io/badge/version-6.1.1-orange)](https://github.com/shopware/platform/releases/tag/v6.1.1)

## Installing

Using yarn:

```bash
$ yarn add @shopware-pwa/shopware-6-client
```


## Usage example

A simple example of how to use API client to connect with your Shopware instance.

In main project file setup connection informations.
```js
import {setup} from "@shopware-pwa/shopware-6-client"

setup({
  endpoint: 'https://address-to-my-shopware-instance.com',
  accessToken: 'myaccesstoken'
})
```

And then anywhere in your project, you can use API Client methods:

```js
import { getCategories } from "@shopware-pwa/shopware-6-client"

// later in component

async mounted() {
  this.categories = await getCategories();
}
```