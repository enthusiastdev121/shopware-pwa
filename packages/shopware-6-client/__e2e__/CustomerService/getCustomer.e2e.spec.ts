import { update, login, getCustomer } from "@shopware-pwa/shopware-6-client";

describe("shopware-6-client - E2E - CustomerService - getCustomer", () => {
  beforeEach(() => {
    update({ contextToken: "" });
  });

  it("should get current customer info", async () => {
    await login({ username: "test.e2e@test.pl", password: "password" });
    const result: any = await getCustomer();
    result.lastLogin = "mockedDate";
    result.updatedAt = "mockedDate";
    expect(result).toMatchSnapshot();
  });

  it("should return null when there is not user logged in", async () => {
    const result = await getCustomer();
    expect(result).toMatchSnapshot();
  });
});
