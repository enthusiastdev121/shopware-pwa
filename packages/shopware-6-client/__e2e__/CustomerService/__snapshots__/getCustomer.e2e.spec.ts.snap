// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shopware-6-client - E2E - CustomerService - getCustomer should get current customer info 1`] = `
Object {
  "_uniqueIdentifier": "d5ba879ae22f4dcd8f881046dbcedc41",
  "active": true,
  "activeBillingAddress": Object {
    "_uniqueIdentifier": "c03217761a5847c19ea3d9801dd2bb7d",
    "additionalAddressLine1": null,
    "additionalAddressLine2": null,
    "city": "TestTown",
    "company": null,
    "country": Object {
      "_uniqueIdentifier": "38245a84c3d5425b8bac97fc845b5ddd",
      "active": true,
      "createdAt": "2019-10-08T09:41:15+00:00",
      "customFields": null,
      "customerAddresses": null,
      "displayStateInRegistration": false,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": null,
          "versionId": null,
        },
        "internal_mapping_storage": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": null,
          "versionId": null,
        },
      },
      "forceStateInRegistration": false,
      "id": "38245a84c3d5425b8bac97fc845b5ddd",
      "iso": "PL",
      "iso3": "POL",
      "name": "Poland",
      "orderAddresses": null,
      "position": 10,
      "salesChannelDefaultAssignments": null,
      "salesChannels": null,
      "shippingAvailable": true,
      "states": null,
      "taxFree": false,
      "taxRules": null,
      "translated": Object {
        "customFields": Array [],
        "name": "Poland",
      },
      "translations": null,
      "updatedAt": null,
      "versionId": null,
    },
    "countryId": "38245a84c3d5425b8bac97fc845b5ddd",
    "countryState": null,
    "countryStateId": null,
    "createdAt": "2020-01-01T20:14:14+00:00",
    "customFields": null,
    "customer": null,
    "customerId": "d5ba879ae22f4dcd8f881046dbcedc41",
    "department": null,
    "extensions": Object {
      "foreignKeys": Object {
        "_uniqueIdentifier": null,
        "createdAt": null,
        "extensions": Array [],
        "translated": Array [],
        "updatedAt": null,
        "versionId": null,
      },
    },
    "firstName": "test",
    "id": "c03217761a5847c19ea3d9801dd2bb7d",
    "lastName": "e2e",
    "phoneNumber": null,
    "salutation": null,
    "salutationId": "c370eb5cd1df4d4dbcc78f055b693e79",
    "street": "teststreet",
    "title": null,
    "translated": Array [],
    "updatedAt": null,
    "vatId": null,
    "versionId": null,
    "zipcode": "10-900",
  },
  "activeShippingAddress": Object {
    "_uniqueIdentifier": "c03217761a5847c19ea3d9801dd2bb7d",
    "additionalAddressLine1": null,
    "additionalAddressLine2": null,
    "city": "TestTown",
    "company": null,
    "country": Object {
      "_uniqueIdentifier": "38245a84c3d5425b8bac97fc845b5ddd",
      "active": true,
      "createdAt": "2019-10-08T09:41:15+00:00",
      "customFields": null,
      "customerAddresses": null,
      "displayStateInRegistration": false,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": null,
          "versionId": null,
        },
        "internal_mapping_storage": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": null,
          "versionId": null,
        },
      },
      "forceStateInRegistration": false,
      "id": "38245a84c3d5425b8bac97fc845b5ddd",
      "iso": "PL",
      "iso3": "POL",
      "name": "Poland",
      "orderAddresses": null,
      "position": 10,
      "salesChannelDefaultAssignments": null,
      "salesChannels": null,
      "shippingAvailable": true,
      "states": null,
      "taxFree": false,
      "taxRules": null,
      "translated": Object {
        "customFields": Array [],
        "name": "Poland",
      },
      "translations": null,
      "updatedAt": null,
      "versionId": null,
    },
    "countryId": "38245a84c3d5425b8bac97fc845b5ddd",
    "countryState": null,
    "countryStateId": null,
    "createdAt": "2020-01-01T20:14:14+00:00",
    "customFields": null,
    "customer": null,
    "customerId": "d5ba879ae22f4dcd8f881046dbcedc41",
    "department": null,
    "extensions": Object {
      "foreignKeys": Object {
        "_uniqueIdentifier": null,
        "createdAt": null,
        "extensions": Array [],
        "translated": Array [],
        "updatedAt": null,
        "versionId": null,
      },
    },
    "firstName": "test",
    "id": "c03217761a5847c19ea3d9801dd2bb7d",
    "lastName": "e2e",
    "phoneNumber": null,
    "salutation": null,
    "salutationId": "c370eb5cd1df4d4dbcc78f055b693e79",
    "street": "teststreet",
    "title": null,
    "translated": Array [],
    "updatedAt": null,
    "vatId": null,
    "versionId": null,
    "zipcode": "10-900",
  },
  "addresses": null,
  "affiliateCode": null,
  "autoIncrement": 244,
  "birthday": null,
  "campaignCode": null,
  "company": null,
  "createdAt": "2020-01-01T20:14:14+00:00",
  "customFields": null,
  "customerNumber": "10242",
  "defaultBillingAddress": Object {
    "_uniqueIdentifier": "c03217761a5847c19ea3d9801dd2bb7d",
    "additionalAddressLine1": null,
    "additionalAddressLine2": null,
    "city": "TestTown",
    "company": null,
    "country": Object {
      "_uniqueIdentifier": "38245a84c3d5425b8bac97fc845b5ddd",
      "active": true,
      "createdAt": "2019-10-08T09:41:15+00:00",
      "customFields": null,
      "displayStateInRegistration": false,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": null,
          "versionId": null,
        },
        "internal_mapping_storage": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": null,
          "versionId": null,
        },
      },
      "forceStateInRegistration": false,
      "id": "38245a84c3d5425b8bac97fc845b5ddd",
      "iso": "PL",
      "iso3": "POL",
      "name": "Poland",
      "position": 10,
      "shippingAvailable": true,
      "states": null,
      "taxFree": false,
      "translated": Object {
        "customFields": Array [],
        "name": "Poland",
      },
      "translations": null,
      "updatedAt": null,
      "versionId": null,
    },
    "countryId": "38245a84c3d5425b8bac97fc845b5ddd",
    "countryState": null,
    "countryStateId": null,
    "createdAt": "2020-01-01T20:14:14+00:00",
    "customFields": null,
    "customer": null,
    "customerId": "d5ba879ae22f4dcd8f881046dbcedc41",
    "department": null,
    "extensions": Object {
      "foreignKeys": Object {
        "_uniqueIdentifier": null,
        "createdAt": null,
        "extensions": Array [],
        "translated": Array [],
        "updatedAt": null,
        "versionId": null,
      },
    },
    "firstName": "test",
    "id": "c03217761a5847c19ea3d9801dd2bb7d",
    "lastName": "e2e",
    "phoneNumber": null,
    "salutation": null,
    "salutationId": "c370eb5cd1df4d4dbcc78f055b693e79",
    "street": "teststreet",
    "title": null,
    "translated": Array [],
    "updatedAt": null,
    "vatId": null,
    "versionId": null,
    "zipcode": "10-900",
  },
  "defaultBillingAddressId": "c03217761a5847c19ea3d9801dd2bb7d",
  "defaultPaymentMethod": Object {
    "_uniqueIdentifier": "cad2356ee68c473ca7da517d0ad32de1",
    "active": true,
    "availabilityRule": null,
    "availabilityRuleId": null,
    "createdAt": "2019-10-08T09:41:16+00:00",
    "customFields": null,
    "description": "Payment upon receipt of goods.",
    "extensions": Object {
      "foreignKeys": Object {
        "_uniqueIdentifier": null,
        "createdAt": null,
        "extensions": Array [],
        "translated": Array [],
        "updatedAt": null,
        "versionId": null,
      },
      "internal_mapping_storage": Object {
        "_uniqueIdentifier": null,
        "createdAt": null,
        "extensions": Array [],
        "translated": Array [],
        "updatedAt": null,
        "versionId": null,
      },
    },
    "id": "cad2356ee68c473ca7da517d0ad32de1",
    "media": null,
    "mediaId": null,
    "name": "Cash on delivery",
    "pluginId": null,
    "position": 1,
    "translated": Object {
      "customFields": Array [],
      "description": "Payment upon receipt of goods.",
      "name": "Cash on delivery",
    },
    "translations": null,
    "updatedAt": null,
    "versionId": null,
  },
  "defaultPaymentMethodId": "cad2356ee68c473ca7da517d0ad32de1",
  "defaultShippingAddress": Object {
    "_uniqueIdentifier": "c03217761a5847c19ea3d9801dd2bb7d",
    "additionalAddressLine1": null,
    "additionalAddressLine2": null,
    "city": "TestTown",
    "company": null,
    "country": Object {
      "_uniqueIdentifier": "38245a84c3d5425b8bac97fc845b5ddd",
      "active": true,
      "createdAt": "2019-10-08T09:41:15+00:00",
      "customFields": null,
      "displayStateInRegistration": false,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": null,
          "versionId": null,
        },
        "internal_mapping_storage": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": null,
          "versionId": null,
        },
      },
      "forceStateInRegistration": false,
      "id": "38245a84c3d5425b8bac97fc845b5ddd",
      "iso": "PL",
      "iso3": "POL",
      "name": "Poland",
      "position": 10,
      "shippingAvailable": true,
      "states": null,
      "taxFree": false,
      "translated": Object {
        "customFields": Array [],
        "name": "Poland",
      },
      "translations": null,
      "updatedAt": null,
      "versionId": null,
    },
    "countryId": "38245a84c3d5425b8bac97fc845b5ddd",
    "countryState": null,
    "countryStateId": null,
    "createdAt": "2020-01-01T20:14:14+00:00",
    "customFields": null,
    "customer": null,
    "customerId": "d5ba879ae22f4dcd8f881046dbcedc41",
    "department": null,
    "extensions": Object {
      "foreignKeys": Object {
        "_uniqueIdentifier": null,
        "createdAt": null,
        "extensions": Array [],
        "translated": Array [],
        "updatedAt": null,
        "versionId": null,
      },
    },
    "firstName": "test",
    "id": "c03217761a5847c19ea3d9801dd2bb7d",
    "lastName": "e2e",
    "phoneNumber": null,
    "salutation": null,
    "salutationId": "c370eb5cd1df4d4dbcc78f055b693e79",
    "street": "teststreet",
    "title": null,
    "translated": Array [],
    "updatedAt": null,
    "vatId": null,
    "versionId": null,
    "zipcode": "10-900",
  },
  "defaultShippingAddressId": "c03217761a5847c19ea3d9801dd2bb7d",
  "doubleOptInConfirmDate": null,
  "doubleOptInEmailSentDate": null,
  "doubleOptInRegistration": false,
  "email": "test.e2e@test.pl",
  "extensions": Object {
    "foreignKeys": Object {
      "_uniqueIdentifier": null,
      "createdAt": null,
      "extensions": Array [],
      "translated": Array [],
      "updatedAt": null,
      "versionId": null,
    },
  },
  "firstLogin": null,
  "firstName": "test",
  "group": null,
  "groupId": "cfbd5018d38d41d8adca10d94fc8bdd6",
  "guest": false,
  "hash": null,
  "id": "d5ba879ae22f4dcd8f881046dbcedc41",
  "language": null,
  "languageId": "2fbb5fe2e29a4d70aa5854ce7ce3e20b",
  "lastLogin": "mockedDate",
  "lastName": "e2e",
  "lastOrderDate": null,
  "lastPaymentMethod": null,
  "lastPaymentMethodId": null,
  "newsletter": false,
  "orderCount": 0,
  "orderCustomers": null,
  "productReviews": null,
  "promotions": null,
  "recoveryCustomer": null,
  "remoteAddress": "10.30.0.0",
  "salesChannel": null,
  "salesChannelId": "98432def39fc4624b33213a56b8c944d",
  "salutation": Object {
    "_uniqueIdentifier": "c370eb5cd1df4d4dbcc78f055b693e79",
    "createdAt": "2019-10-08T09:41:13+00:00",
    "displayName": "Mr.",
    "extensions": Object {
      "foreignKeys": Object {
        "_uniqueIdentifier": null,
        "createdAt": null,
        "extensions": Array [],
        "translated": Array [],
        "updatedAt": null,
        "versionId": null,
      },
      "internal_mapping_storage": Object {
        "_uniqueIdentifier": null,
        "createdAt": null,
        "extensions": Array [],
        "translated": Array [],
        "updatedAt": null,
        "versionId": null,
      },
    },
    "id": "c370eb5cd1df4d4dbcc78f055b693e79",
    "letterName": "Dear Mr.",
    "salutationKey": "mr",
    "translated": Object {
      "displayName": "Mr.",
      "letterName": "Dear Mr.",
    },
    "translations": null,
    "updatedAt": null,
    "versionId": null,
  },
  "salutationId": "c370eb5cd1df4d4dbcc78f055b693e79",
  "tags": null,
  "title": null,
  "translated": Array [],
  "updatedAt": "mockedDate",
  "versionId": null,
}
`;

exports[`shopware-6-client - E2E - CustomerService - getCustomer should return null when there is not user logged in 1`] = `null`;
